integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/endpoint/test/HttpEndpointTestCase.java

```diff
@@ -91,89 +83,64 @@ public void testHttpEndpoint() throws Exception {
-                "         <!--0 to 1 occurrence-->\n" +
-                "         <xs:name xmlns:xs=\"http://axis2.apache.org\">" + studentName + "</xs:name>\n" +
-                "         <!--0 or more occurrences-->\n" +
-                "         <xs:subjects xmlns:xs=\"http://axis2.apache.org\">testAutomation</xs:subjects>\n" +
-                "      </ns:student>\n" +
-                "   </p:addStudent>";
-
-        StringReader sendData = new StringReader(addStudentData);
-        StringWriter responseData = new StringWriter();
-        URL restURL = new URL((getProxyServiceURLHttp("postEPProxy")) + "/students/");
-        HttpURLConnectionClient.sendPostRequest(sendData, restURL, responseData, "application/xml");
-
-        assertTrue(responseData.toString().contains(studentName), "response doesn't contain the expected output");
+
+        String createCustomerData =
+                "<createCustomer>\n" + "<id>" + customerId + "</id>\n" + "<name>" + customerName + "</name>\n"
+                        + "</createCustomer>";
+        StringReader customerData = new StringReader(createCustomerData);
+        StringWriter postResponseData = new StringWriter();
+        URL postRestURL = new URL((getApiInvocationURL(CUSTOMER_API_CONTEXT)) + RESOURCE_CONTEXT);
+        System.out.println();
+        HttpURLConnectionClient.sendPostRequest(customerData, postRestURL, postResponseData, "application/xml");
+        assertTrue(postResponseData.toString().contains(customerName),
+                "response doesn't contain the expected output but contains: " + postResponseData.toString());
+
     }
 
-    @Test(groups = {"wso2.esb"}, description = "HTTP Endpoint GET test: RESTful", priority = 6, enabled = false)
+    @Test(groups = { "wso2.esb" }, description = "HTTP Endpoint GET test: RESTful", priority = 6)
     public void testToGet() throws IOException {
-        //check whether the student is added.
-        String studentGetUri = getProxyServiceURLHttp("getEPProxy") + "/student/" + studentName;
-        HttpResponse getResponse = HttpURLConnectionClient.sendGetRequest(studentGetUri, null);
-
-        assertTrue(getResponse.getData().contains("<ns:getStudentResponse xmlns:ns=\"http://axis2.apache.org\"><ns:return>" +
-                "<ns:age>100</ns:age>" +
-                "<ns:name>" + studentName + "</ns:name>" +
-                "<ns:subjects>testAutomation</ns:subjects>" +
-                "</ns:return></ns:getStudentResponse>"));
+
+        String getRestURI = getApiInvocationURL(CUSTOMER_API_CONTEXT) + RESOURCE_CONTEXT + "/" + customerId;
+        HttpResponse getResponseData = HttpURLConnectionClient.sendGetRequest(getRestURI, null);
+        assertTrue(getResponseData.getData().contains(
+                "<getCustomerResponse xmlns=\"http://ws.apache.org/ns/synapse\"><id>8fa3fc1b-f63c-4b21-8aff-3ac684c74d97</id><name>John</name></getCustomerResponse>"),
+                "Unexpected output received:" + getResponseData.toString());
```

Comment:
Shall we add "<getCustomerResponse xmlns=\"http://ws.apache.org/ns/synapse\"><i...." to a variable as done in the next test

Reply:
In the next test, a variable is added because we are only asserting for the update value property (name of customer), instead of the entire response payload and.Does it make sense to add the entire payload to a variable?

Reply:
Ah got it, for execution, it doesn't make sense, but for readability, I feel it's better if we have variable

Reply:
+1, Fixed with bafd177